measured.go

```diff
@@ -273,12 +273,16 @@ func (mc *Conn) Close() (err error) {
 
 func (mc *Conn) submitTrafficIfNecessary() {
 	now := time.Now()
+	needToSubmit := false
 	mc.submitMutex.Lock()
 	if now.Sub(mc.lastSubmitted) > mc.interval {
-		mc.submitTraffic()
+		needToSubmit = true
 		mc.lastSubmitted = now
 	}
 	mc.submitMutex.Unlock()
+	if needToSubmit {
```

Comment:
@myleshorton I moved the submit out of the locked section.  As mentioned, read locks aren't upgradeable in go, so I'm sticking with a regular lock.